src/pages/DeviceSettings.tsx

```diff
@@ -0,0 +1,368 @@
+                        });
+                        getDevicesFc();
+                        return;
+                    }).catch(err => {
+                        showDialog({
+                            content:"Error updating device meta: "+err,
+                            onAccept:()=>{},
+                            onCancel:()=>{},
+                            hideCloseButton: true,
+                            acceptBtnTitle:"Close",
+                            title:"Error encountered"
+                        });
+                    });
+            }
+            if (device?.name !== thisDevice?.name) {
+                window.wazigate.setDeviceName(thisDevice.id as string, thisDevice.name.toString())
+                    .then(() => {
+                        showDialog({
+                            content:"Device name updated ",
+                            onAccept:()=>{},
+                            onCancel:()=>{},
+                            hideCloseButton: true,
+                            acceptBtnTitle:"Close",
+                            title:"Update successfull."
+                        });
+                        return;
+                    }).catch(err => {
+                        showDialog({
+                            content:"Error updating device name "+err,
+                            onAccept:()=>{},
+                            onCancel:()=>{},
+                            acceptBtnTitle:"Close",
+                            hideCloseButton: true,
+                            title:"Error encountered"
+                        });
+                    });
+            }
+        }
+        setIsEdited(false);
+        setIsEditedCodec(false);
+        getDevicesFc();
+    }
+    useEffect(() => {
+        window.wazigate.getDevice(id as string).then((de) => {
+            if(de.meta.lorawan){
+                setThisDevice({
+                    ...de,
+                    name: de.name,
+                });
+            }else{
+                setThisDevice({
+                    ...de,
+                    name: de.name,
+                    meta: {
+                        ...de.meta,
+                        lorawan: {
+                            devEUI: '',
+                            devAddr: '',
+                            nwkSEncKey: '',
+                            appSKey: '',
+                            profile: "WaziDev",
+                        }
+                    }
+                });
+            }
+        });
+    }, [id]);
+    const handleChangeDeviceCodec = (event: SelectChangeEvent<string>) => {
+        setThisDevice({
+            ...thisDevice as Device,
+            meta: {
+                ...thisDevice?.meta,
+                codec: event.target.value
+            }
+        });
+        setIsEditedCodec(true);
+    };
+    const autoGenerateLoraWANOptions = (title: "devAddr" | "nwkSEncKey" | "appSKey") => {
```

Comment:
What this function does? Please add a comment on top

Reply:
It autogenerates the LoraWAN keys when you place the + icon.

Reply:
Ok.